### ISSUE TITLE
inject does not work in standalone server components

### ISSUE THREAD

**Issue by arsengoian** [nuxt/nuxt#22751]
### Environment

- Operating System: Windows_NT (Windows 11)
- Node Version:     v18.16.1
- Nuxt Version:     3.6.5
- Nitro Version:    2.5.2
- Package Manager:  yarn@1.19.1
- Builder:          vite
- User Config:      devtools, experimental, modules, target
- Runtime Modules:  @pinia/nuxt@0.4.11
- Build Modules:    -

### Reproduction

Please refer to the reproduction:
https://stackblitz.com/edit/github-jvdfpf?file=components%2FServerTest.server.vue

Note that Test.vue and ServerTest.server.vue are the same, however for the second file the injected value is not displayed.

### Describe the bug

The issue is that the provided value cannot be injected (is reset to the default value in standalone server components/ using component islands).

Refer to https://vuejs.org/guide/components/provide-inject.html for expected behavior.

Would be thankful for a fix or workaround that would allow to use standalone server components with provide / inject. Unfortunately, I don't have any insight into what might be causing the issue. This can't be intended behaviour though, right? Because simply server side rendering components works just fine.

Would be happy to help provided anyone has any ideas what can be done 😎

**Comment by danielroe** [comment]
Yes, server components here are rendered 'outside' the tree which is why they don't have access to provide/inject - they need to be able to be rendered on their own.

Why do you need to use provide/inject? Could a Nuxt plugin work in your case?

**Comment by arsengoian** [comment]
Thanks for the reply! Well, it would be really nice to use provide/inject. 

I'm developing a module that would help achieve something like this:

```vue
<MyComponent context="$">
  <MyOtherComponent context = ".items"/>
    <SomeoneElsesComponent/>
        <LibraryItem context="[0]"/>
        <LibraryItem context="[1]"/>
     <SomeoneElsesComponent/>
  </MyOtherComponent/>
<MyComponent/>
```

MyComponent and MyOtherComponent would provide the value up the tree and it would result in a string `$.items[0]` for example, which could be accessed by `LibraryItem`, which would then perform operations with a key in the JSON tree. `LibraryItem` is supposed to be a standalone server component, therefore if provide/inject does not work, I'll have to use full JSON paths for the context prop or write a lot of boilerplate for concatenating props (which wouldn't even work if there's some component I can't modify in the way in the tree, which is likely to happen in my project)

Is there any other workaround that would enaable to achieve something like this? I don't see how I could achieve this with a plugin, since it would just inject the same value to all components, regardless of the tree position

**Comment by nukuutos** [comment]
is it a problem when we do `provide` inside server component and do `inject` inside nested server component?
```
<server-component-with-provide>
    <server-component-with-inject />
</server-component-with-provide>
```
I see a problem when we mix server and client component with `provide` and `inject`. but when there are no mix, no problem should arise.
